mixer/adapter/dogstatsd/config/config.proto

```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+// The `dogstatsd` adapter is designed to deliver Istio metric instances to a
+// listening [DataDog](https://www.datadoghq.com/) agent.
+package adapter.dogstatsd.config;
+
+import "gogoproto/gogo.proto";
+
+option go_package="config";
+
+// Configuration parameter for the DataDog adapter. These params control how mixer telemetry is transformed and sent to a DataDog agent.
+//
+// The adapter assumes that a datadog daemon is running as a sidecar or at some other endpoint that the mixer can reach.
+// Any dimension that is a part of the metric is converted to a tag automatically. The configuration of the DataDog agent/daemon is outside the scope of the adapter.
+message Params {
+    // Address of the dogstatsd server. 
+    // Default: localhost:8125
+    string address = 1;
+
+    // prefix to prepend to all metrics handled by the adapter. Metric "bar" with prefix "foo." becomes "foo.bar" in DataDog.
+    // Default: ""
+    string prefix = 2;
+
+    // UDP buffer size in bytes 
+    // Default: 1432
+    int32 buffer_length = 3;
+
+    // Tags to add to every metric. "global": "tag" becomes "global:tag" in DataDog
+    // Default: []
+    map<string, string> global_tags = 4;
+
+    // Chance that any particular metric is sampled when emitted; can take the range [0, 1].
+    // Default: 1
+    double sample_rate = 5;
+
+    // Describes how to represent this metric in DataDog
+    message MetricInfo {
+        // Name of the metric in DataDog
+        string name = 1;
+        // Describes the type of metric
+        enum Type {
+            // Default Unknown Type
+            UNKNOWN_TYPE = 0;
+            // Increments a DataDog counter
+            COUNTER = 1;
+            // Sets the new value of a DataDog gauge
+            GAUGE = 2;
+            // DISTRIBUTION is converted to a Timing Histogram for metrics with a time unit and a Histogram for all other units
+            DISTRIBUTION = 3;
+        }
+        // The type of metric
+        Type type = 2;
+        // Tags to add to the metric in addition to the dimensions. "tag": "val" becomes "tag:val" in DataDog
+        // Default: []
+        map<string, string> tags = 3;
+    }
+
+    // Map of a specific metric instance name -> info. If a metric's instnace name is not in the map then the metric will not be exported to DataDog.
```

Comment:
It would be preferable to generate when this happens. Otherwise, the operator could easily have data being dumped in the toilet and not know why. So just generate errors when things don't line up.

You can discover exactly what you will be handed at runtime during validation. So you can report errors to the user there, before receiving traffic.